panel/widgets/icon.py

```diff
@@ -25,20 +31,87 @@ class ToggleIcon(Widget):
 
-    _widget_type = _PnToggleIcon
+    _widget_type = _PnClickableIcon
 
-    _rename: ClassVar[Mapping[str, str | None]] = {'name': 'name', 'button_style': None}
+    _rename: ClassVar[Mapping[str, str | None]] = {
+        **TooltipMixin._rename, 'name': 'name',
+    }
+
+    _source_transforms: ClassVar[Mapping[str, str | None]] = {
+        'description': None,
+    }
 
     _stylesheets: ClassVar[List[str]] = [f'{CDN_DIST}css/icon.css']
 
     def __init__(self, **params):
         super().__init__(**params)
 
-    @param.depends("icon", "active_icon", watch=True, on_init=True)
+    @param.depends('icon', 'active_icon', watch=True, on_init=True)
     def _update_icon(self):
         if not self.icon:
             raise ValueError('The icon parameter must not be empty.')
 
         icon_is_svg = self.icon.startswith('<svg')
         if icon_is_svg and not self.active_icon:
-            raise ValueError('The active_icon parameter must not be empty if icon is an SVG.')
+            raise ValueError(
+                'The active_icon parameter must not be empty if icon is an SVG.'
+            )
+
+
+class ToggleIcon(_ClickableIcon, TooltipMixin):
+
+    _widget_type = _PnToggleIcon
+
+
+class ButtonIcon(_ClickableIcon, _ClickButton, TooltipMixin):
+    clicks = param.Integer(default=0, doc="""
+        The number of times the button has been clicked.""")
+
+    toggle_duration = param.Integer(default=75, doc="""
+        The number of milliseconds the active_icon should be shown for
+        and how long the button should be disabled for.""")
+
+    value = param.Boolean(default=False, doc="""
+        Toggles from False to True while the event is being processed.""")
+
+    _widget_type = _PnButtonIcon
+
+    _rename: ClassVar[Mapping[str, str | None]] = {
+        **TooltipMixin._rename, 'name': 'name',
+    }
+
+    _target_transforms: ClassVar[Mapping[str, str | None]] = {
+        'event:button_click': None,
+    }
+
+    def __init__(self, **params):
+        click_handler = params.pop('on_click', None)
+        super().__init__(**params)
+        if click_handler:
+            self.on_click(click_handler)
+
+    def on_click(
+        self, callback: Callable[[param.parameterized.Event], None]
+    ) -> param.parameterized.Watcher:
+        """
+        Register a callback to be executed when the button is clicked.
+
+        The callback is given an `Event` argument declaring the number of clicks
+
+        Arguments
+        ---------
+        callback: (Callable[[param.parameterized.Event], None])
+            The function to run on click events. Must accept a positional `Event` argument
+
+        Returns
+        -------
+        watcher: param.Parameterized.Watcher
+          A `Watcher` that executes the callback when the MenuButton is clicked.
+        """
+        return self.param.watch(callback, 'clicks', onlychanged=False)
+
+    def _process_event(self, event: param.parameterized.Event) -> None:
```

Comment:
This seems like your problem, you need to implement the logic:

```
        self.param.trigger('value')
        self.clicks += 1
```

Reply:
Note that the type annotation here is also incorrect. If you could please fix this one and the one on `Button._process_event` both should use:

```python
from bokeh.events import ButtonClick
...
    def _process_event(self, event: ButtonClick) -> None:
```

Reply:
Just wondering is there a benefit to implement clicks += 1 on the Python side vs TS?

Reply:
Yes, as I said, property changes are throttled and potentially discarded if they are queued up, events are not.